src/AWS.Deploy.CLI/Commands/CommandFactory.cs

```diff
@@ -106,43 +109,49 @@ public Command BuildRootCommand()
         {
             // Name is important to set here to show correctly in the CLI usage help.
             // Either dotnet-aws or dotnet aws works from the CLI. System.Commandline's help system does not like having a space with dotnet aws.
-            var rootCommand = new RootCommand {
+            var rootCommand = new RootCommand
+            {
                 Name = "dotnet-aws",
                 Description = "The AWS .NET deployment tool for deploying .NET applications on AWS."
             };
 
-            rootCommand.Add(BuildDeployCommand());
-            rootCommand.Add(BuildListCommand());
-            rootCommand.Add(BuildDeleteCommand());
-            rootCommand.Add(BuildDeploymentProjectCommand());
-            rootCommand.Add(BuildServerModeCommand());
+            lock(this)
```

Comment:
why are you not using s_locker?

Reply:
@ganeshnj can weigh in on this

Reply:
RootCommand is an enumerable which is not thread safe. This is needed to make the insertion thread safe. It must be `s_locker`. `this` will not allow all threads to lock.

https://github.com/dotnet/command-line-api/issues/1258

Reply:
Fixed